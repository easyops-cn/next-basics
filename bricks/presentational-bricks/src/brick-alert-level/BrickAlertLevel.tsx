import React from "react";

import {
  BrickValueMapping,
  MappingValue,
  Color,
} from "../brick-value-mapping/BrickValueMapping";
import { AlertLevel } from "./index";

const mapping: { [key: string]: MappingValue } = {
  0: {
    text: "通知",
    color: Color.blue,
  },
  1: {
    text: "警告",
    color: Color.orange,
  },
  2: {
    text: "紧急",
    color: Color.red,
  },
  info: {
    text: "通知",
    color: Color.blue,
  },
  warning: {
    text: "警告",
    color: Color.orange,
  },
  critical: {
    text: "紧急",
    color: Color.red,
  },
};

interface BrickAlertLevelProps {
  value: AlertLevel;
}

export function BrickAlertLevel(
  props: BrickAlertLevelProps
): React.ReactElement {
  return (
    <BrickValueMapping
      value={props.value}
      mapping={mapping}
    ></BrickValueMapping>
  );
}
