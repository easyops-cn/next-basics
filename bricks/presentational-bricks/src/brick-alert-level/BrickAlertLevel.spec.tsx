import React from "react";
import { shallow } from "enzyme";
import { BrickAlertLevel } from "./BrickAlertLevel";

describe("BrickAlertLevel", () => {
  it("should work", () => {
    const wrapper = shallow(<BrickAlertLevel value={1} />);
    expect(wrapper).toMatchSnapshot();
  });
});
