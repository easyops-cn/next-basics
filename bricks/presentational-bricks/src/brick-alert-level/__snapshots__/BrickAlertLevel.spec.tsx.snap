// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrickAlertLevel should work 1`] = `
<BrickValueMapping
  mapping={
    Object {
      "0": Object {
        "color": "blue",
        "text": "通知",
      },
      "1": Object {
        "color": "orange",
        "text": "警告",
      },
      "2": Object {
        "color": "red",
        "text": "紧急",
      },
      "critical": Object {
        "color": "red",
        "text": "紧急",
      },
      "info": Object {
        "color": "blue",
        "text": "通知",
      },
      "warning": Object {
        "color": "orange",
        "text": "警告",
      },
    }
  }
  value={1}
/>
`;
